source/gui/addonGui.py

```diff
@@ -5,128 +5,344 @@
+		).format(**bundle.manifest)
+		# Translators: A message asking the user to confirm that they are aware that this addon has not been tested.
+		confirmAddonNotTestedMessage = _("I understand that this add-on has not been tested with my version of NVDA")
+
+		return ConfirmAddonInstallDialog(
+			parent=self,
+			# Translators: Title for message asking if the user really wishes to install an Addon.
+			title=_("Warning: Untested Add-on Installation"),
+			message=confirmInstallMessage,
+			userConfirmationMessage=confirmAddonNotTestedMessage,
+			showAddonInfoFunction=lambda: self._showAddonInfo(bundle.manifest)
+		).ShowModal()
+
+	def _showConfirmAddonInstallDialog(self, bundle):
+		# Translators: A message asking the user if they really wish to install an addon.
+		confirmInstallMessage = _(
+			"Are you sure you want to install this add-on?\n"
+			"Only install add-ons from trusted sources.\n"
+			"Addon: {summary} {version}"
+		).format(**bundle.manifest)
+
+		return ConfirmAddonInstallDialog(
+			parent=self,
+			# Translators: Title for message asking if the user really wishes to install an Addon.
+			title=_("Add-on Installation"),
+			message=confirmInstallMessage,
+			showAddonInfoFunction=lambda: self._showAddonInfo(bundle.manifest)
+		).ShowModal()
+
 	def installAddon(self, addonPath, closeAfter=False):
 		try:
 			try:
-				bundle=addonHandler.AddonBundle(addonPath)
+				bundle = addonHandler.AddonBundle(addonPath)
 			except:
-				log.error("Error opening addon bundle from %s"%addonPath,exc_info=True)
-				# Translators: The message displayed when an error occurs when opening an add-on package for adding. 
-				gui.messageBox(_("Failed to open add-on package file at %s - missing file or invalid file format")%addonPath,
+				log.error("Error opening addon bundle from %s" % addonPath, exc_info=True)
+				gui.messageBox(
+					# Translators: The message displayed when an error occurs when opening an add-on package for adding.
+					_("Failed to open add-on package file at %s - missing file or invalid file format") % addonPath,
 					# Translators: The title of a dialog presented when an error occurs.
 					_("Error"),
-					wx.OK | wx.ICON_ERROR)
+					wx.OK | wx.ICON_ERROR
+				)
 				return
-			# Translators: A message asking the user if they really wish to install an addon.
-			if gui.messageBox(_("Are you sure you want to install this add-on? Only install add-ons from trusted sources.\nAddon: {summary} {version}\nAuthor: {author}").format(**bundle.manifest),
-				# Translators: Title for message asking if the user really wishes to install an Addon.
-				_("Add-on Installation"),
-				wx.YES|wx.NO|wx.ICON_WARNING)!=wx.YES:
+
+			if not addonVersionCheck.hasAddonGotRequiredSupport(bundle):
+				self._showAddonRequiresNVDAUpdateDialog(bundle)
 				return
-			bundleName=bundle.manifest['name']
+
+			if not addonVersionCheck.isAddonTested(bundle):
+				if wx.YES != self._showAddonUntestedDialog(bundle):

```

Comment:
`if self._showAddonUntestedDialog(bundle) == wx.NO:` would be more in line with other code.